aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -84,9 +84,10 @@ static void apply_triangular_solve(Tensor& A, Tensor& B, bool upper, bool transp
   auto A_mat_stride = matrixStride(A);
   auto B_mat_stride = matrixStride(B);
   auto batch_size = batchCount(A);
-  auto n = cuda_int_cast(A.size(-2), "n");
+  auto m = cuda_int_cast(A.size(-2), "m");
+  auto n = cuda_int_cast(A.size(-1), "n");
   auto nrhs = cuda_int_cast(B.size(-1), "nrhs");
-  auto lda = std::max<int>(1, n);
+  auto lda = std::max<int>(1, m);
```

Comment:
Are we testing this case? If not, it'd be good to add tests for the rectangular cases.

Reply:
We don't test it. It's indirectly tested now with the lstsq/gels using it. The public interface through `torch.triangular_solve` doesn't allow rectangular inputs.